java/common/src/test/scala/org/typelevel/otel4s/java/ContextStorageSuite.scala

```diff
@@ -0,0 +1,43 @@
+package org.typelevel.otel4s.java
+
+import cats.effect.IO
+import cats.effect.IOLocal
+import cats.effect.Resource
+import cats.effect.std.Dispatcher
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.ContextKey
+import io.opentelemetry.context.ContextStorage
+import java.util.logging._
+
+object Run extends cats.effect.IOApp.Simple {
+
+  val key = ContextKey.named[String]("test")
+
+  def run =
+    Dispatcher.parallel[IO].use { dispatcher =>
```

Comment:
Actually if you use a `sequential` dispatcher here instead of `parallel` you might have more success with this approach. `sequential` runs all submitted effects on the same fiber, so `IOLocal` state would persist.

Reply:
It's super nasty anyway, because the constant wrapper (obliterating any other "middleware") has to run before `Context` is classloaded, or it logs a FINE (i.e., not on by default) error.  I think your PR may enable a more proper SPI approach.